.github/workflows/release-rc.yml

```diff
@@ -1,18 +1,48 @@
 name: Release Armada components - RC
 
 on:
-  push:
+  workflow_run:
+    types: [completed]
+    workflows: [CI]
     branches:
-      - main
       - master
 
 permissions:
   contents: write
 
 jobs:
+  validate:
+    if: github.event.workflow_run.event == 'push' && github.event.workflow_run.conclusion == 'success' && github.repository_owner == 'armadaproject'
+    name: "Validate revision"
+    runs-on: ubuntu-22.04
+
+    steps:
+      - name: "Checkout"
+        uses: "actions/checkout@v3"
+        with:
+          fetch-depth: 0
+
+      - name: Validate ref
+        run: |
+          ref='${{ github.event.workflow_run.head_branch }}'
+          sha='${{ github.event.workflow_run.head_sha }}'
+          case $ref in
+            master)
+              [ $(git branch --contains=$sha master | wc -l) -eq 1 ] &&
+              [ $(git rev-list --count $sha..master) -le 2 ]
+              ;;
+            *)
+              false
+              ;;
+          esac
```

Comment:
you don't need the case anymore, this could be simplified to
```
          [ "$ref" == "master" ] &&
          [ $(git branch --contains=$sha master | wc -l) -eq 1 ] &&
          [ $(git rev-list --count $sha..master) -le 2 ]
```